src/contracts/atlas/Atlas.sol

```diff
@@ -221,4 +220,41 @@ contract Atlas is Test, Factory {
         } 
         return IExecutionEnvironment(executionEnvironment).validateUserCall(userMetaTx);
     }
+
+    function metacallSimulation(
+        ProtocolCall calldata protocolCall,
+        UserCall calldata userCall,
+        SearcherCall[] calldata searcherCalls,
+        Verification calldata verification
+    ) external payable {
+        if (!metacall(protocolCall, userCall, searcherCalls, verification)) {
+            revert("ERR-S01 NoAuctionWinner");
+        }
+        revert("ERR-S00 SimulationPassed");
+    }
+
+    function testSearcherCalls(
+        ProtocolCall calldata protocolCall,
+        UserCall calldata userCall,
+        SearcherCall[] calldata searcherCalls,
+        Verification calldata verification
+    ) external payable returns (bool auctionWon) {
+        try this.metacallSimulation{value: msg.value}(protocolCall, userCall, searcherCalls, verification) {}
+        catch (bytes memory revertData) {
+            for (uint256 i; i < revertData.length-4;) {
```

Comment:
Might be cleaner to use a custom error's selector (bytes4) as a switch, similar to the SearcherWrapper

https://github.com/Polygon-Fast-Lane/atlas/blob/33a0ef6bbff7cbbf0220ba632457d08c60020edc/src/contracts/atlas/SearcherWrapper.sol#L36-L56

Reply:
Done in https://github.com/Polygon-Fast-Lane/atlas/pull/15/commits/d82f00d81f8ec88a26548adcaeb2574f7be03c84